Source/Laser/LaserParticleContainer.cpp

```diff
@@ -678,6 +692,9 @@ LaserParticleContainer::update_laser_particle (WarpXParIter& pti,
             // Calculate the velocity according to the amplitude of E
             const Real sign_charge = (pwp[i]>0) ? 1 : -1;
             const Real v_over_c = sign_charge * tmp_mobility * amplitude[i];
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(amrex::Math::abs(v_over_c) < amrex::Real(1.),
+                            "Error: calculated laser particle velocity greater than c."
+                            "Make sure the laser wavelength and amplitude are accurately set.");
```

Comment:
Is it a good idea to add an assert inside this GPU kernel? If not, we can remove the ALWAYS so that we only do the assert in debug mode.

Reply:
Just inquired about this: the GPU version of `AMREX_ALWAYS_ASSERT_WITH_MESSAGE` will call `assert(0)` in case of failure, which is dependent on the (non-)definition of `NDEBUG`. So in other words: `assert` will only be present in debug builds, `Print()` will always be there.

This is different on CPU, where the `ALWAYS_` macros call `abort()` on failure. This is called for all build modes, independent of the `NDEBUG` define.

In both cases, it will always print something if the test fails, independent of `NDEBUG`/build mode.

Reply:
Thanks! I've tested on GPU and indeed the error message is printed (which can get quite scary if all the threads print the same message :D ) but the run continues until the segfault.

Performance-wise there indeed seems to be no noticeable difference. (in any case the laser particle push is only 0.1-0.15% of the total time in the 3D laser injection test, which has 1 laser and no particles).

Reply:
Thank you for the details. Yes, the many message are a bit sub-ideal, but at least it will crash and have some kind of notice before it.

Performance sounds very reasonable.